ppmclibs/tinycv_impl.cc

```diff
@@ -316,6 +316,24 @@ double getPSNR(const Mat& I1, const Mat& I2)
     return 10.0 * log10(signal / (noise * noise));
 }
 
+/*!
+ * \brief Creates OpenCV's threads upfront.
+ *
+ * This allows one to take control over spawning these threads, e.g. to block signals.
+ *
+ * \remarks
+ * - This function does a color space conversion to provoke the thread creation because there
+ *   seems to be no designated public API for that.
+ * - Apparently the image size must exceed a certain threshold to trigger multithreading. Hence
+ *   this function does not use an 1x1 px image.
+ */
+void create_opencv_threads()
+{
+    const Mat input = Mat::zeros(400, 400, CV_8UC3);
+    Mat output = Mat::zeros(400, 400, CV_8UC3);
+    cvtColor(input, output, cv::COLOR_BGR2GRAY);
```

Comment:
Seems like this is not good enough. It creates 7 additional threads on my machine with 8 cores but on a production-like machine with 32 cores no further threads are created. Not sure how to create the OpenCV threads upfront in a reliable way.

Reply:
Updated the PR. It seems `parallel_for_` is actually a public API. The new `create_opencv_threads` function now works reliably with any number of threads.

Reply:
Just following the code I appreciate the extensive documentation you added. Pretty neat trick to drive up the thread count like that.